14_scroll_slidein/app.js

```diff
@@ -0,0 +1,37 @@
+function debounce(func, wait = 10, immediate = true) {
+  var timeout;
+  return function () {
+    var context = this,
+      args = arguments;
+    var later = function () {
+      timeout = null;
+      if (!immediate) func.apply(context, args);
+    };
+    var callNow = immediate && !timeout;
+    clearTimeout(timeout);
+    timeout = setTimeout(later, wait);
+    if (callNow) func.apply(context, args);
+  };
+}
+
+const sliderImages = document.querySelectorAll(".slide-in");
+
+function checkSlide() {
+  sliderImages.forEach((sliderImage) => {
+    const amountScrolled = 50;
+    const slideInAt =
+      window.scrollY + window.innerHeight - sliderImage.clientHeight / 2;
+    console.log(slideInAt);
+    const imageBottom = sliderImage.offsetTop + sliderImage.clientHeight;
+    console.log(imageBottom);
+    const isHalfShown = slideInAt > sliderImage.offsetTop;
```

Comment:
Math that involves scroll and window can be tricky.

My best advice to you is to have a pen and a paper by your side and put these calculations into sketches. Draw them to help you understand it. It's my trick every single time I need to deal with scroll animations.